File: Java/src/d1klasaobjekt/nikadaneraditiovako/Dokument.java
package d1klasaobjekt.nikadaneraditiovako;

public class Dokument {

    int sifra;
    String brojDokumenta;
    double iznos;

    void obrada(){
        System.out.println("Obrađujem " + sifra + " - " + brojDokumenta + " - " + iznos);
    }

}

File: Kotlin/src/main/kotlin/t4polimorfizam/Dokument.kt
package t4polimorfizam

/**
 * @author Matija Skrletović
 **/

abstract class Dokument(val sifra: Int, val brojDokumenta: String, val iznos: Double, val izradio: Osoba? = null) : Any() {

    abstract fun obradiDokument()
    abstract fun getOpis(): String?

    override fun toString(): String {
        return "Dokument{ sifra=$sifra, brojDokumenta='$brojDokumenta', iznos=$iznos}"
    }

}